ops/model.py

```diff
@@ -213,20 +199,48 @@ def __iter__(self):
         return iter(self._data)
 
     def __getitem__(self, relation_name):
+        is_peer = relation_name in self._peers
         relation_list = self._data[relation_name]
         if relation_list is None:
             relation_list = self._data[relation_name] = []
-            for relation_id in self._backend.relation_ids(relation_name):
-                relation_list.append(Relation(relation_name, relation_id, self._our_unit, self._backend, self._cache))
+            for rid in self._backend.relation_ids(relation_name):
+                relation = Relation(relation_name, rid, is_peer, self._our_unit, self._backend, self._cache)
+                relation_list.append(relation)
         return relation_list
 
+    def _get_unique(self, relation_name, relation_id=None):
+        if not isinstance(relation_id, (int, type(None))):
+            raise ModelError(f'relation name {relation_id} must be int or None not {type(relation_id).__name__}')
+        if relation_id is not None:
```

Comment:
I'd suggest moving the first `if` inside the `None` if, as that simplifies it: that is,
```python
if relation_id is not None:
    if not isinstance(relation_id, int):
        raise ModelError('...')
    # etc
```

Reply:
Thanks, that indeed simplifies it.

https://github.com/canonical/operator/pull/107/commits/6a058cb1cd04cb7f3d4ea38c01fd13a04d95cd03